### ISSUE TITLE
The serializer for type 'System.Collections.ObjectModel.ReadOnlyCollection`1 could not be found.

### ISSUE THREAD

**Issue by michidk** [wojtpl2/ExtendedXmlSerializer#152]
Hello,

Today I looked back into my old game engine project which is using your xml serialize. It improved a lot and a lot of features which I wished for, are now implemented. Great! And congratulations for the release of v2 🎉 !

XMLSerializer is runs perfect and I already use the new features. But I have still one issue:
I have a class `GameObject : IEnumerable<GameObject>` [same issue without inheriting the IEnumberable] which has the following member: `    [XmlElement]
        private readonly List<GameObject> children = new List<GameObject>();`

The serialization of the class works if I remove this variable. If I don't remove it, the following error is thrown:
```
System.InvalidOperationException: 'The serializer for type 'System.Collections.ObjectModel.ReadOnlyCollection1[BrokenEngine.SceneGraph.GameObject]' could not be found.  Please ensure that the type is a valid type can be activated.  The default behavior requires an empty public constructor on the (non-abstract) class to activate.'
```

The GameObject class can be found here: https://github.com/michidk/BrokenEngine/blob/c48c522d97609a069558665d5de5243512a69bfc/BrokenEngine/SceneGraph/GameObject.cs

What do I have to change to make it work?

**Comment by michidk** [comment]
I just noticed that I do have ReadyOnlyCollections in my class. but they are annotated with [XmlIgnore]!
```
       [XmlIgnore]
        public ReadOnlyCollection<GameObject> Children => children.AsReadOnly();
        [XmlIgnore]
        public ReadOnlyCollection<Component> Components => components.AsReadOnly();

        private Vector3 localPosition;
        private Quaternion localRotation = Quaternion.Identity;
        private Vector3 localScale = Vector3.One;
[...]

        [XmlElement]
        private readonly List<GameObject> children = new List<GameObject>();
       // [XmlElement]
        private readonly List<Component> components = new List<Component>();
```

Renaming them fixed the issue. So I guess I just found a bug?

**Comment by Mike-EEE** [comment]
Ah thanks for reporting this @michidk.  I have been slammed with Real World work but I should start getting issues from this repo attended to starting tomorrow.  I will let you know what I find. :)

**Comment by Mike-EEE** [comment]
Next stop on my apology tour. 😄 I just checked in a fix for this bug @michidk and sorry it took so long!  There was an issue in how we were resolving fields and properties.  If they had the same name then it would select the property instead.  This is now fixed and can be accessed using our preview feed immediately:
https://www.myget.org/F/wojtpl2/api/v3/index.json

We're also aiming to have a new version deployed officially to NuGet later on today. 👍
